### DISCUSSION TITLE
Stenographer missing- stopping threads

### DISCUSSION THREAD

**Discussion by luke879** [Security-Onion-Solutions/securityonion#9761] [Unsupported Versions]
Hi,

Hoping for some assistance with Stenograph showing as "missing" using so-status. Not sure where to go with this. Appreciate any help!

2023-02-09T22:39:58.444279Z T:b602d7 [aio.cc:190] Opening packet file /tmp/stenographer520628931/PKT0/.1675982398444036: 4
2023-02-09T22:39:58.956910Z T:b602d7 [stenotype.cc:537] Thread 0 stats: MB=1 secs=1.5e-05 MBps=66666.7 packets=946 blocks=1 polls=1 drops=0 drop%=0
panic: runtime error: slice bounds out of range [:4768] with capacity 519

goroutine 18 [running]:
github.com/golang/leveldb/table.(*blockIter).Next(0xc0003fc180, 0xc0002a8600)
        /home/mreeves/rpmbuild/BUILD/stenographer-e8db1ee8f6b62790ebc8081e464dcedd7c961766/src/github.com/golang/leveldb/table/reader.go:125 +0x3f4
github.com/golang/leveldb/table.block.seek(0xc0002e6d80, 0x474, 0x479, 0xbd6d20, 0x16fc060, 0xc0003e8041, 0x1, 0x1, 0xc000069b00, 0x79ed4d, ...)
        /home/mreeves/rpmbuild/BUILD/stenographer-e8db1ee8f6b62790ebc8081e464dcedd7c961766/src/github.com/golang/leveldb/table/reader.go:85 +0x190
github.com/golang/leveldb/table.(*Reader).find(0xc00035bae0, 0xc0003e8041, 0x1, 0x1, 0x0, 0x0, 0x16fc060, 0xc000069b00)
        /home/mreeves/rpmbuild/BUILD/stenographer-e8db1ee8f6b62790ebc8081e464dcedd7c961766/src/github.com/golang/leveldb/table/reader.go:369 +0xef
github.com/golang/leveldb/table.(*Reader).Get(0xc00035bae0, 0xc0003e8041, 0x1, 0x1, 0x0, 0x1, 0x100000000000003, 0x0, 0x30, 0xc0004450b0)
        /home/mreeves/rpmbuild/BUILD/stenographer-e8db1ee8f6b62790ebc8081e464dcedd7c961766/src/github.com/golang/leveldb/table/reader.go:337 +0x8d
github.com/google/stenographer/indexfile.NewIndexFile(0xc0004450b0, 0x30, 0xc0002824b0, 0x3, 0xaff380, 0x3)
        /home/mreeves/rpmbuild/BUILD/stenographer-e8db1ee8f6b62790ebc8081e464dcedd7c961766/src/github.com/google/stenographer/indexfile/indexfile.go:67 +0x23d
github.com/google/stenographer/blockfile.NewBlockFile(0xc000445050, 0x30, 0xc0002824b0, 0xc000445050, 0x30, 0x1)
        /home/mreeves/rpmbuild/BUILD/stenographer-e8db1ee8f6b62790ebc8081e464dcedd7c961766/src/github.com/google/stenographer/blockfile/blockfile.go:63 +0x14b
github.com/google/stenographer/thread.(*Thread).trackNewFile(0xc0002ac0a0, 0xc00043a230, 0x10, 0x10, 0x16fe000)
        /home/mreeves/rpmbuild/BUILD/stenographer-e8db1ee8f6b62790ebc8081e464dcedd7c961766/src/github.com/google/stenographer/thread/thread.go:172 +0xc9
github.com/google/stenographer/thread.(*Thread).syncFilesWithDisk(0xc0002ac0a0)
        /home/mreeves/rpmbuild/BUILD/stenographer-e8db1ee8f6b62790ebc8081e464dcedd7c961766/src/github.com/google/stenographer/thread/thread.go:139 +0x2af
github.com/google/stenographer/thread.(*Thread).SyncFiles(0xc0002ac0a0)
        /home/mreeves/rpmbuild/BUILD/stenographer-e8db1ee8f6b62790ebc8081e464dcedd7c961766/src/github.com/google/stenographer/thread/thread.go:363 +0x45
github.com/google/stenographer/env.(*Env).syncFiles(0xc0002c8000)
        /home/mreeves/rpmbuild/BUILD/stenographer-e8db1ee8f6b62790ebc8081e464dcedd7c961766/src/github.com/google/stenographer/env/env.go:267 +0x50
github.com/google/stenographer/env.(*Env).callEvery(0xc0002c8000, 0xc000284210, 0x37e11d600)
        /home/mreeves/rpmbuild/BUILD/stenographer-e8db1ee8f6b62790ebc8081e464dcedd7c961766/src/github.com/google/stenographer/env/env.go:180 +0x83
created by github.com/google/stenographer/env.New
        /home/mreeves/rpmbuild/BUILD/stenographer-e8db1ee8f6b62790ebc8081e464dcedd7c961766/src/github.com/google/stenographer/env/env.go:131 +0x3f8
2023-02-09T22:40:21.460088Z T:b682e7 [stenotype.cc:437] Got signal 15, stopping threads

**Comment by InfosecGoon** [comment]
What is the output from this command on the host running Stenographer?

sudo salt-call pillar.get steno

**Reply by luke879** [reply]
sudo salt-call pillar.get steno
local:
    ----------
    enabled:
        True

**Reply by InfosecGoon** [reply]
Okay, doesn't look like there's anything weird in the configuration - is there disk space available in /nsm on the sensor?

**Comment by luke879** [comment]
devtmpfs         65745728         0  65745728   0% /dev
tmpfs            65762108       820  65761288   1% /dev/shm
tmpfs            65762108    151024  65611084   1% /run
tmpfs            65762108         0  65762108   0% /sys/fs/cgroup
/dev/md126p2   4206116864 165345508 821740540  17% /
/dev/md126p2   4206116864 165345508 821740540  17% /home
/dev/nvme0n1p2    1038336    164052    874284  16% /boot
/dev/nvme0n1p1     204580     11468    193112   6% /boot/efi
tmpfs            13152424         0  13152424   0% /run/user/940
tmpfs            13152424         0  13152424   0% /run/user/1000

Looks good so far. I'm wondering if it's a permissions issue? This is a pretty fresh install.

**Reply by InfosecGoon** [reply]
That doesn't look like a standard disk layout -- what version of SO are you running? Is it a network install? What's the base OS?

**Comment by luke879** [comment]
[root@itsec-so steno]# cat config

{
  "Threads": [
    { "PacketsDirectory": "/nsm/pcap", "IndexDirectory": "/nsm/pcapindex", "MaxD                                                                                                                                                             irectoryFiles": 30000, "DiskFreePercentage": 10 }
  ]
  , "StenotypePath": "/usr/bin/stenotype"
  , "Interface": "bond0"
  , "Port": 1234
  , "Host": "127.0.0.1"
  , "Flags": ["-v", "--blocks=2048", "--preallocate_file_mb=4096", "--aiops=128"                                                                                                                                                             , "--uid=stenographer", "--gid=stenographer"]
  , "CertPath": "/etc/stenographer/certs"
}

**Comment by luke879** [comment]
Hi Matt-

Dell BIOS RAID 0 (2TB ea) with two spinners and one NVME for the OS (512GB).

Thanks,

Tom

On Feb 15, 2023, at 5:57 AM, Matthew Gracie ***@***.***> wrote:

﻿

That doesn't look like a standard disk layout -- what version of SO are you running? Is it a network install? What's the base OS?

**Comment by luke879** [comment]
Fresh install / partitioned from SO ISO.

On Feb 15, 2023, at 7:14 AM, Thomas Lukins ***@***.***> wrote:

﻿
Hi Matt-

Dell BIOS RAID 0 (2TB ea) with two spinners and one NVME for the OS (512GB).

Thanks,

Tom

On Feb 15, 2023, at 5:57 AM, Matthew Gracie ***@***.***> wrote:

﻿

That doesn't look like a standard disk layout -- what version of SO are you running? Is it a network install? What's the base OS?
